homeassistant/components/alexa_local_control.py

```diff
@@ -0,0 +1,527 @@
+# this string is required per the SSDP spec
+SSDP_RESPONSE = """HTTP/1.1 200 OK
+CACHE-CONTROL: max-age=60
+EXT:
+LOCATION: http://{0}:{1}/description.xml
+SERVER: FreeRTOS/6.0.5, UPnP/1.0, IpBridge/0.1
+ST: urn:schemas-upnp-org:device:basic:1
+USN: uuid:Socket-1_0-221438K0100073::urn:schemas-upnp-org:device:basic:1
+
+"""
+
+DESCRIPTION_XML_RESPONSE = """<?xml version="1.0" encoding="UTF-8" ?>
+<root xmlns="urn:schemas-upnp-org:device-1-0">
+<specVersion>
+<major>1</major>
+<minor>0</minor>
+</specVersion>
+<URLBase>http://{0}:{1}/</URLBase>
+<device>
+<deviceType>urn:schemas-upnp-org:device:Basic:1</deviceType>
+<friendlyName>HA-Echo ({0})</friendlyName>
+<manufacturer>Royal Philips Electronics</manufacturer>
+<manufacturerURL>http://www.philips.com</manufacturerURL>
+<modelDescription>Philips hue Personal Wireless Lighting</modelDescription>
+<modelName>Philips hue bridge 2015</modelName>
+<modelNumber>BSB002</modelNumber>
+<modelURL>http://www.meethue.com</modelURL>
+<serialNumber>1234</serialNumber>
+<UDN>uuid:2f402f80-da50-11e1-9b23-001788255acc</UDN>
+<presentationURL>index.html</presentationURL>
+<iconList>
+<icon>
+<mimetype>image/png</mimetype>
+<height>48</height>
+<width>48</width>
+<depth>24</depth>
+<url>hue_logo_0.png</url>
+</icon>
+<icon>
+<mimetype>image/png</mimetype>
+<height>120</height>
+<width>120</width>
+<depth>24</depth>
+<url>hue_logo_3.png</url>
+</icon>
+</iconList>
+</device>
+</root>
+"""
+
+CONF_HOST_IP = 'host_ip'
+CONF_LISTEN_PORT = 'listen_port'
+CONF_OFF_MAPS_TO_ON_DOMAINS = 'off_maps_to_on_domains'
+CONF_EXPOSE_BY_DEFAULT = 'expose_by_default'
+CONF_EXPOSED_DOMAINS = 'exposed_domains'
+
+ATTR_ECHO = 'echo'
+ATTR_ECHO_FRIENDLY_NAME = 'echo_friendly_name'
+
+DEFAULT_LISTEN_PORT = 8300
+DEFAULT_OFF_MAPS_TO_ON_DOMAINS = ['script', 'scene']
+DEFAULT_EXPOSE_BY_DEFAULT = True
+DEFAULT_EXPOSED_DOMAINS = [
+    'switch', 'light', 'script', 'scene', 'group', 'input_boolean',
+    'media_player'
+]
+
+
+def setup(hass, config):
+    """Activate the alexa_local_control component."""
+    from flask import Flask
+
+    app = Flask(__name__)
+    app.url_map.strict_slashes = False
+
+    view = HueBridgeView(hass, config)
+    view.set_up_routes(app)
+
+    def start_listener(event):
+        """Start listening for UPNP/SSDP requests."""
+        upnp_listener = UPNPResponderThread(
+            view.host_ip_addr, view.listen_port)
+        upnp_listener.start()
+
+    def start_web_server():
+        """Start the Hue bridge API web server."""
+        from cherrypy import wsgiserver
+
+        dispatcher = wsgiserver.WSGIPathInfoDispatcher({'/': app})
+        server = wsgiserver.CherryPyWSGIServer(
+            ('0.0.0.0', view.listen_port), dispatcher)
+
+        server.start()
+
+    hass.bus.listen_once(EVENT_HOMEASSISTANT_START, start_listener)
+    threading.Thread(target=start_web_server, daemon=True, name=DOMAIN).start()
+
+    return True
+
+
+class HueBridgeView(object):

```

Comment:
Assuming you can move to `http`, this should be [`HomeAssistantView`](https://github.com/robbiet480/home-assistant/blob/09d531b3b95e7943df53233d1abcb40d7e87de94/homeassistant/components/http.py#L426).